src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,112 @@
+ */
+public class Parser {
+    /**
+     * Parses the input string and returns the matching Command object that executes the command.
+     *
+     * @param input String representing user input.
+     * @return Command object.
+     * @throws DukeException if the input is not a valid command.
+     */
+    public static Command parseCommand(String input) throws DukeException {
+        if (input.isBlank()) {
+            throw new DukeException(ErrorMessage.MESSAGE_INVALID_INPUT);
+        }
+        String command = input.split(" ")[0];
+        String[] args = input.split(" ", 2);
+        switch (command) {
+        case "list":
+            return new ListCommand();
+            // Fallthrough
+        case "delete":
+            if (args.length == 1) {
+                throw new DukeException(ErrorMessage.MESSAGE_UNKNOWN_INDEX);
+            }
+            return new DeleteCommand(Integer.parseInt(args[1]));
+            // Fallthrough
+        case "mark":
+            if (args.length == 1) {
+                throw new DukeException(ErrorMessage.MESSAGE_UNKNOWN_INDEX);
+            }
+            return new MarkCommand(Integer.parseInt(args[1]));
+            // Fallthrough
+        case "unmark":
+            if (args.length == 1) {
+                throw new DukeException(ErrorMessage.MESSAGE_UNKNOWN_INDEX);
+            }
+            return new UnmarkCommand(Integer.parseInt(args[1]));
+            // Fallthrough
+        case "deadline":
+            if (args.length == 1) {
+                throw new DukeException(ErrorMessage.MESSAGE_UNKNOWN_DESC);
+            }
+            String[] deadline = args[1].split("/by", 2);
+            if (deadline.length == 1) {
+                throw new DukeException(ErrorMessage.MESSAGE_UNKNOWN_DATE);
+            }
+            return new AddCommand(new Deadline(deadline[0], parseDate(deadline[1])));
+            // Fallthrough
+        case "event":
+            if (args.length == 1) {
+                throw new DukeException(ErrorMessage.MESSAGE_UNKNOWN_DESC);
+            }
+            String[] event = args[1].split("/at", 2);
+            if (event.length == 1) {
+                throw new DukeException(ErrorMessage.MESSAGE_UNKNOWN_DATE);
+            }
+            return new AddCommand(new Event(event[0], parseDate(event[1])));
+            // Fallthrough
+        case "todo":
+            if (args.length == 1) {
+                throw new DukeException(ErrorMessage.MESSAGE_UNKNOWN_DESC);
+            }
+            return new AddCommand(new Todo(args[1]));
+            // Fallthrough
+        case "find":
+            if (args.length == 1) {
+                throw new DukeException(ErrorMessage.MESSAGE_UNKNOWN_KEYWORD);
+            }
+            return new FindCommand(args[1]);
+        case "bye":
+            return new ExitCommand();
+            // Fallthrough
+        default:
+            throw new DukeException(ErrorMessage.MESSAGE_UNKNOWN_COMMAND);
+        }
+    }
```

Comment:
Would it be clearer if the method was broken down to smaller methods to be called at each case?

Reply:
I did it the same way as @jxt00 as well, but I think using switch-case makes it easier to tell that the entire block is solely trying to call the appropriate commands.

Reply:
Yep, as the method is longer than 30 LOC, I extracted out the parsing parts into smaller methods.